gourmet/gtk_extras/LinkedTextView.py

```diff
@@ -97,7 +120,7 @@ def event_after(self,
             try:
                 start, end = buffer.get_selection_bounds()
             except ValueError:
-                # If there is nothing selected, None is return
+                # If there is nothing selected, None is returned
```

Comment:
Can we actually return `None` then? Also, it doesn't look like `None` is returned. I only see `return False` and only few code paths that do not have a `return`.

Reply:
We're not returning `None`, it's bad phrasing in the comment:

`buffer.get_selection_bounds()` may return `None` if there is no selection, (as opposed to a tuple of `TextIter`.

This `try/except/else` statement could be rewritten as something like:
```python
            # Do not follow links if the user has selected something
            selection = buffer.get_selection_bounds()
            if selection is not None:
                start, end = selection
                if start.get_offset() != end.get_offset():
                    # ie. not a real selection, but the position of the cursor
                    # in widget
                    return False
```

This function basically does a check that we're not doing text selection, (which is triggered by the same events [button release].)

Now that I think of it, the selection check could be done before:

```python
    def event_after(self,
                    text_view: 'LinkedTextView',
                    event: Gdk.Event) -> bool:
        """Handle mouse clicks on time links."""
        _, button = event.get_button()

        # Check for selection
        selecting = False

        buffer = text_view.get_buffer()
        selection = buffer.get_selection_bounds()
        if selection is not None:
            if selection[0].get_offset() != selection[1].get_offset():
                selecting = True

        # Check for a left mouse click (as set by the system, not hardware).
        if event.type == Gdk.EventType.BUTTON_RELEASE and button == 1 and not selecting:
            x, y = text_view.window_to_buffer_coords(Gtk.TextWindowType.WIDGET,
                int(event.x), int(event.y))
            _, itr = text_view.get_iter_at_location(x, y)
            self.follow_if_link(text_view, itr)

        return False  # Don't process the event further
```

Reply:
I always dislike the mental parsing when an `if` is just to choose between values. Why not the following?

`selecting = bool(selection is not None and (selection[0].get_offset() != selection[1].get_offset()))`?

Reply:
That also works, yes!